component/pyroscope/scrape/target.go

```diff
@@ -45,10 +45,13 @@ const (
 type Target struct {
 	// Labels before any processing.
 	discoveredLabels labels.Labels
-	// Any labels that are added to this target and its metrics.
-	labels labels.Labels
+	// Any privateLabels that are added to this target and its metrics.
+	privateLabels labels.Labels
+	labels        labels.Labels
```

Comment:
It seems like we don't want the disjoint private/public labels, because I think it would break, for example:
```
t.appender.Append(context.Background(), t.privateLabels, []*pyroscope.RawSample{{RawProfile: b}});
```
I think you'll want to have `allLabels` and `publicLabels`, no need for private ones. I can take a look tomorrow.